### DISCUSSION TITLE
Open Graph Meta Tags

### DISCUSSION THREAD

**Discussion by corytron** [chipzoller/hugo-clarity#155] [Q&A] [answered]
Hi, while setting my site I noticed there are some default meta tags related to open graph for sharing your site on social. For example.

`<meta property="og:type" content="website">`
`<meta name="description" content="A theme based on VMware&#39;s Clarity Design System for publishing technical blogs with Hugo.">`
`<meta property="og:url" content="https://vcf.static421.io/" />`
`<meta property="og:title" content="" />`
`<meta property="og:description" content="A theme based on VMware&#39;s Clarity Design System for publishing technical blogs with Hugo." />`
`<meta property="og:image" content="https://vcf.static421.io/images/og.png" />`

in config.toml you expose a parameter: `fallBackOgImage = "images/og.png"`

which does alter `og:image` but how do I adjust `og:title` and `og:description` ? They do not seem to be using my title parameter from config.toml

Any help is very much appreciated.

-Cory

**Comment by chipzoller** [comment]
@onweru Would this be a regression from a more recent commit? IIRC, these should be picked up automatically.

**Reply by onweru** [reply]
I don't think so, that regression would show up here view-source:https://themes.gohugo.io/theme/hugo-clarity/

**Comment by corytron** [comment, marked as answer]
after I wrote this I actually found that setting author and description in content/_index.md allows me to get all the OG meta tags done correctly. But it seems like setting it in config.toml wasn't working.

**Reply by chipzoller** [reply]
I'm unclear here. Is this a theme problem (or something we can provide) or is this a bit of user customization being applied? @onweru is way more familiar with OG than I.

**Reply by corytron** [reply]
I think it's likely a theme issue, since I doubt you guys added open graph meta tag support to the template. I'm still learning exactly what Hugo provides so I appreciate your help here.

**Reply by onweru** [reply]
> this a bit of user customization being applied?

Yes, it's on the user's end. 

>  since I doubt you guys added open graph meta tag support to the template

Doubt shouldn't be at play here. [This template](https://github.com/chipzoller/hugo-clarity/blob/master/layouts/partials/opengraph.html) covers just that. 

Going back to the @corytron's first description:

> They do not seem to be using my title parameter from config.toml

It would really help to see your config.toml. My hunch is that you're doing something wrong.

**Reply by corytron** [reply]
[Here it is](https://github.com/corytron/vcf/blob/master/config.toml)

The gist of the issue, as I've experienced it, is that the `description`  and `author` field, set under `[params]` in config.toml don't have any impact on the OG meta tags. Only after changing them in [_index.md](https://github.com/corytron/vcf/blob/master/content/_index.md) seems to get the OG meta tags to update.

**Reply by onweru** [reply]
I cloned your repo and run `hugo server`. The meta tags seem alright to me. Attaching the source screenshots (one for the homepage and one for an article)

<img width="1440" alt="Screen Shot 2021-03-23 at 10 24 49 PM" src="https://user-images.githubusercontent.com/16350351/112206191-b2b84a00-8c26-11eb-95e0-976e763cf68d.png">
<img width="1440" alt="Screen Shot 2021-03-23 at 10 25 18 PM" src="https://user-images.githubusercontent.com/16350351/112206231-bea40c00-8c26-11eb-9a4d-0226fd415aba.png">

**Reply by onweru** [reply]
You're right though, the `author` value isn't applied on the meta tags.

**Reply by corytron** [reply]
Thanks for looking at this,

I just committed an update to content/_index.md to make it clearer if the data is coming from that file or config.toml

If the description is coming from _index.md it'll say `If you're reading this its coming from content/_index.md and not config.toml` in the source. And the author will say "Hugo Authors" instead of me. It still seems like the Author and description from config.toml is not being picked up when I rebuilt the site. Now I'm using netlify and hugo 0.80.0 I wonder if it could be a change in hugo?

**Reply by onweru** [reply]
Oh, I see. That's purely intentional i.e the theme is supposed to work that way. Your `description` from [your homepage](https://github.com/corytron/vcf/blob/master/content/_index.md#L4) will override the `description` from the [config](https://github.com/corytron/vcf/blob/ed8f45d71948db85bc3c74fbd9f2df85733bfa65/config.toml#L52). If you prefer the description in your config to be used, simply delete  [line 4 from your _index.md](https://github.com/corytron/vcf/blob/master/content/_index.md#L4)

**Reply by onweru** [reply]
Please see variables that are overridable on [these tables](https://github.com/chipzoller/hugo-clarity#global-parameters)

**Reply by corytron** [reply]
I see, that makes sense. I think what was throwing me off was that _index.md was prepopulated with data so when I set the `description` in config.toml I wasn't seeing any change. Thanks for clarifying

**Reply by onweru** [reply]
No worries. Happy publishing!
